### PR TITLE
Fix avs crash

### PR DESCRIPTION
**Pull request by David-Henner**
### Issues

Joining a group call from a second device crashes 

### Solutions

Use CallParticipant instead of UUID & update the logic responsible for updating the participants & participants states

### CODE REVIEW

File: Wire-iOS Tests/CallHapticsControllerTests.swift [hunk 1]
```diff
@@ -31,20 +31,30 @@ fileprivate class MockCallHapticsGenerator: CallHapticsGeneratorType {
     }
 }
 
-final class CallHapticsControllerTests: XCTestCase {
+final class CallHapticsControllerTests: ZMSnapshotTestCase {
```
Comment (billypchan) [inline Wire-iOS Tests/CallHapticsControllerTests.swift]: These are no snapshot tests below, any reason for changing this?

Comment (David-Henner) [reply Wire-iOS Tests/CallHapticsControllerTests.swift]: Nice catch!
I've used it to get easy access to the managed object context, and forgot to update it in the end

Comment (johnxnguyen) [reply Wire-iOS Tests/CallHapticsControllerTests.swift]: Discussed 1:1. For now it's ok to leave it like this. We made a ticket to refactor the test classes to provide a context for normal unit tests.

Comment (billypchan) [reply Wire-iOS Tests/CallHapticsControllerTests.swift]: You may use `CoreDataFixture` class which extracted the methods in `ZMSnapshotTestCase `

**Pull request merged**